### ISSUE TITLE
InfluxDB 0.9 crashed when enable collectd and scan that port

### ISSUE THREAD

**Issue by ldkingvivi** [influxdb/influxdb#3008]
the old issue has been closed , so please re-open this one

you can reproduce this on the latest version right now, which is 0.9

Here's the only conf I made from default conf:
```
[collectd]
enabled = true
bind-address = ":25826"
# database = ""
typesdb = "/root/collectd/share/collectd/types.db"
```

netstat verify it's running on the port:

```
tcp 0 0 :::8083 :::* LISTEN 28488/influxd
tcp 0 0 :::8086 :::* LISTEN 28488/influxd
tcp 0 0 :::8088 :::* LISTEN 28488/influxd
udp 0 0 :::25826 :::* 28488/influxd
```

running command to scan the UDP port 25826:

`nc -vnzu 127.0.0.1 25826`

then it crashed

**Comment by neonstalwart** [comment]
ref #2772

**Comment by otoolep** [comment]
This does indeed crash InfluxDB. The panic is in the gollect library, and it's not clear how best to deal with this. We could either patch it or add a call to `recover` to our code.

It occurs because only 1 byte is available in the UDP packet, but gollectd expects 2.

https://github.com/kimor79/gollectd/blob/cf6dec97343244b5d8a5485463675d42f574aa2d/gollectd.go#L125

**Comment by otoolep** [comment]
https://github.com/kimor79/gollectd/issues/6 opened against gollectd.

**Comment by otoolep** [comment]
https://github.com/kimor79/gollectd/pull/7 will fix gollectd. If that is not merged, we'll have no choice but to fork gollectd.

**Comment by kimor79** [comment]
Merged the gollectd patched

**Comment by otoolep** [comment]
Fixed 0.9.1.

**Comment by otoolep** [comment]
Thanks @kimor79
